### DISCUSSION TITLE
Adding Custom Field values from device_type to device

### DISCUSSION THREAD

**Discussion by Zombie-Toad** [netbox-community/netbox#18059] [Q&A] [answered]
We are running a Printserver to install printers on our client machines. With every printer model there is a preconfigured driver and a driver template for different OS. 
In netbox, I added a custom field for my drivers and assigned it to my driver_types and devices.
When I create a new device with device_type=printer, the custom field is not added to the new device.
Is there any option to add custom field values from device_type to a newly created device?
I assumed that it would work like the interface option on device_type which creates the configured interfaces as soon as a new device of that type is created.

**Comment by candlerb** [comment, marked as answer]
Custom fields on the device type belong to the device type only. They are not copied to instances of the device. In fact, Device and DeviceType, being separate models, have their own definitions for custom fields.

Similarly, custom fields on an interface template (i.e. an interface which is part of the device type definition) are not copied to instances of the interface on the created device.

**Reply by Zombie-Toad** [reply]
OK, but is there any option to only show the custom fields on specific device_types? Like some sort of dynamic forms?

I know that I can create custom scripts, but I could not figure out how to implement an input field for custom fields.

**Reply by candlerb** [reply]
> is there any option to only show the custom fields on specific device_types? Like some sort of dynamic forms?

There is not. ISTR it has been requested before in some old issue/discussion.

**Reply by Zombie-Toad** [reply]
Too bad, then I try to figure out how to put custom fields into my custom script.
Thanks for your help.
